java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertVarToExplicitType.java

```diff
@@ -152,16 +155,25 @@ private static boolean hasDiagnosticErrors(CompilationInfo info, Tree tree) {
     private static boolean isValidType(HintContext ctx) {
         TreePath treePath = ctx.getPath();
         TreePath initTreePath = ctx.getVariables().get("$init");  //NOI18N
+        TypeMirror variableTypeMirror = ctx.getInfo().getTrees().getElement(treePath).asType();
 
         if (initTreePath.getLeaf().getKind() == Tree.Kind.NEW_CLASS) {
             NewClassTree nct = ((NewClassTree) initTreePath.getLeaf());
             if (nct.getClassBody() != null) {
                 return false;
             }
+        } else if (initTreePath.getLeaf().getKind() == Tree.Kind.METHOD_INVOCATION) {
```

Comment:
Not sure if this check is correct:
-why look at the structure of the initializer when we have the variable type? If the variable type contains anonymous, that is a problem, if it does not, all is fine (i.e. why not simple look at variableTypeMirror?). What if the code is like:
var v1 = new Object() {};
var v2 = v1; //v2 is anonymous as well, right?
-what about cases like:
var v = List.of(new Object() {});
?

Reply:
Fixed these usecases. Need to handle usecase var v = {new Object()} in the new error hint for array initializer. I'm trying to push that change in same PR.

Reply:
Fixed